lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,350 @@
+import stream from 'stream';
+
+import { errors } from 'arsenal';
+import kineticlib from 'kinetic-js';
+
+const HEADER_SZ = 9;
+
+class Kinetic {
+
+    /**
+     * @constructor
+     * @param {Object.<string>[]} drives - the list of drives
+     * @returns {Kinetic} - to allow for a functional style.
+     */
+    constructor(drives) {
+        this._sockets = [];
+        this._drives = drives;
+        this._sockets = this._drives.map(drive => ({
+            drive,
+            clusterVersion: 0,
+            connectionID: 0,
+            sock: this._newSocket(),
+            _request: false,
+            _requestGet: false,
+            _events: new EventEmitter(),
+        }));
+        this._chunkTab = [];
+        this._count = 0;
+        this._lexiKey = '0';
+        this.connect();
+    }
+
+    /**
+     * Connect the sockets to the drives
+     * @return {undefined}
+     */
+    connect() {
+        this._sockets.forEach((socket, index) => {
+            socket.sock.on('connect', () => {
+                this._bindReadable(socket, index);
+            }).connect(socket.drive);
+        });
+    }
+
+    getSocketIndex(drive) {
+        for (let i = 0; i < this._sockets.length; i++) {
+            if (this._sockets[i]._request === false
+                && this._sockets[i]._requestGet === false
+                && this._sockets[i].drive === drive) {
+                return i;
+            }
+        }
+
+        /*
+         * TODO : create new socket if no one is available
+         */
+
+        // this._sockets.push(
+        //     {
+        //         drive,
+        //         clusterVersion: 0,
+        //         connectionID: 0,
+        //         sequence: this._sequence,
+        //         sock: this._newSocket(),
+        //         _request: false,
+        //         _requestGet: false,
+        //         _events: new EventEmitter(),
+        //     });
+        // const index = this._sockets.length - 1;
+        // const sock = this._sockets[index];
+        // sock.sock.on('connect', () => {
+        //     this._bindReadable(sock, index);
+        //     return index;
+        // }).connect(sock.drive);
+    }
+
+    _bindReadable(socket, index) {
+        socket.sock.on('readable', err => {
+            if (socket._requestGet === false) {
+                this._parsePDU(socket, (err1, pdu) => {
+                    if (!err1){
+                        this._parseRequest(err || err1, pdu, index);
+                    }
+                });
+            } else {
+                this._endGet(index);
+            }
+        });
+    }
+
+    _parsePDU(socket, callback) {
+        const header = socket.sock.read(HEADER_SZ);
+        if (header !== null) {
+            const protobufSize = header.readInt32BE(1);
+            const rawData = socket.sock.read(protobufSize);
+            if (rawData !== null) {
+                const pdu = new kineticlib.PDU(
+                    Buffer.concat([header, rawData]));
```

Comment:
You can pass the final length to ```Buffer.concat()``, so it doesn't have to calculate it and so is a bit faster.
```
Buffer.concat([header, rawData], header.length + rawData.length)
```